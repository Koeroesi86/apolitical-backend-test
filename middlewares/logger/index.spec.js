const { createRequest, createResponse } = require("node-mocks-http");
const loggerMiddleware = require('./index')

describe('loggerMiddleware middleware', () => {
  const logger = { log: jest.fn() };
  const middleware = loggerMiddleware(logger);

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should call next', async () => {
    const request = createRequest({
      method: 'GET'
    });
    const response = createResponse();
    const next = jest.fn();

    await middleware(request, response, next);
    response.status(200);
    response.send('test');

    expect(next).toHaveBeenCalled();
  });

  it('should call logger.log', async () => {
    const request = createRequest({
      method: 'GET'
    });
    const response = createResponse({
      eventEmitter: require('events').EventEmitter
    });
    const next = jest.fn();

    await middleware(request, response, next);
    response.status(200);
    response.end('test');

    expect(logger.log).toHaveBeenCalled();
    expect(logger.log.mock.calls).toMatchSnapshot();
  });
})
