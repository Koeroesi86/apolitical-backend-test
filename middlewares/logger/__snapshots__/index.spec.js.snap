// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loggerMiddleware middleware should call logger.log 1`] = `
Array [
  Array [
    Object {
      "count": 0,
      "duration": 0,
      "level": "info",
      "message": "Response sent",
      "method": "GET",
      "status": 200,
    },
  ],
]
`;
